src/index.js

```diff
@@ -19,16 +20,89 @@ function createElement(tagName, props, ...children) {
 
   return element;
 }
+const numberKeys = ['1', '2', '3', '4', '5', '6', '7', '8', '9', '0'];
+const operatorKeys = ['+', '-', '*', '/', '='];
+
+function render({
+  displayNumber, operation, tmpResult, lastInput,
+}) {
+  const props = {
+    displayNumber, operation, tmpResult, lastInput,
+  };
+  // console.log(props);
+
+  function calcuation(value) {
+    let result = 0;
+    switch (operation) {
+    case '+':
+      result = tmpResult + parseInt(displayNumber, 10);
+      break;
+    case '-':
+      result = tmpResult - parseInt(displayNumber, 10);
+      break;
+    case '*':
+      result = tmpResult * parseInt(displayNumber, 10);
+      break;
+    case '/':
+      result = tmpResult / parseInt(displayNumber, 10);
+      break;
+    default:
+      break;
+    }
+    // console.log('계산 결과 확인 ! ', result);
+    render({
+      displayNumber: result, tmpResult: value === '=' ? 0 : result, operation: value === '=' ? '' : value, lastInput: value,
+    });
+  }
+
+  function handleClick(value) { // 통합 클릭 함수
+    // 연산 조작 버튼을 클릭한 경우
+    if (operatorKeys.some((key) => key === value)) {
+      if (value === '=') {
+        calcuation(value);
+        return;
+      }
+      if (operation == null) {
+        render({
+          ...props, operation: value, tmpResult: parseInt(displayNumber, 10), lastInput: value,
+        });
+        return;
+      }
+      console.log('연속해서 숫자와 연산 숫자 연산을 누를때');
+      calcuation(value);
+      return;
+    }
+    // 숫자 패드 클릭한 경우
+    if (lastInput === '=') { // 새로운 계산을 시작한 경우
+      render({ displayNumber: value, tmpResult: parseInt(value, 10), lastInput: value });
+      return;
+    }
+    if (operatorKeys.some((key) => key === lastInput)) { // 이전에 눌린 키가 연산관련키라면,
+      render({ ...props, displayNumber: value, lastInput: value });
+      return;
+    }
+    render({ ...props, displayNumber: displayNumber + value, lastInput: value });
+  }
 
-function render() {
   const element = (
     <div>
       <p>간단 계산기</p>
+      <p>{displayNumber || 0}</p>
+      <div>
+        {numberKeys.map((key) => (
+          <button type="button" id={`num${key}`} onClick={() => { handleClick(key); }}>{key}</button>
+        ))}
+      </div>
+      <div>
+        {operatorKeys.map((key) => (
+          <button type="button" id={`operation${key}`} onClick={() => { handleClick(key); }}>{key}</button>
+        ))}
+      </div>
```

Comment:
`key`라는 키워드는 html 태그들의 attibute중 하나이기 때문에 가능하면 다른 네이밍을 사용하는게 좋을 것 같아요~

```suggestion
        {numbers.map((number) => (
          <button type="button" id={`num- ${number}`} onClick={() => { handleClick(number) }}>{number}</button>
        ))}
      </div>
      <div>
        {operators.map((operator) => (
          <button type="button" id={`operation-${operator}`} onClick={() => { handleClick(operator) }}>{operator}</button>
        ))}
      </div>
```

참고: 
- https://ko.reactjs.org/docs/lists-and-keys.html

Reply:
네 수정하겠습니다!